lib/blockscore/base.rb

```diff
@@ -4,14 +4,28 @@ module BlockScore
   class Base
     extend Connection
 
-    attr_reader :attributes
 
     def initialize(options = {})
+      @loaded = !(options.instance_of?(Proc))
       @attributes = options
     end
 
+    # potential issues here
+    def attributes
+      return @attributes if @loaded
+      force!
+      @attributes
+    end
+
+    def force!
+      @attributes = @attributes.call().attributes
+      @loaded = true
```

Comment:
Style/MethodCallParentheses: Do not use parentheses for method calls with no arguments.